compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -607,34 +620,59 @@ class ComposeScene internal constructor(
             previousPressOwner.processPointerInput(event)
             return
         }
+        val position = event.pointers.first().position
         forEachOwnerReversed { owner ->
-            if (owner.isHovered(event)) {
-                // Stop once the position of in bounds of the owner
+
+            // If the position of in bounds of the owner - send event to it and stop processing
+            if (owner.isInBounds(position)) {
                 owner.processPointerInput(event)
                 pressOwner = owner
-                return@processPress
+                return
             }
-            owner.onClickOutside?.invoke()
+
+            // Input event is out of bounds - send click outside notification
+            owner.onOutsidePointerEvent?.invoke(event)
+
+            // if the owner is in focus, do not pass the event to underlying owners
             if (owner == focusedOwner) {
-                // Stop if it's in focus, do not pass the event to hovered owner
-                return@processPress
+                return
             }
         }
     }
 
     private fun processRelease(event: PointerInputEvent) {
-        // Send Release to pressOwner even if is not hovered or under focused.
+        // Send Release to pressOwner even if is not hovered or under focused
         pressOwner?.processPointerInput(event)
+
+        // Process further only if last pointer was released or had pressOwner
+        if (event.isAnyPointerDown || pressOwner != null) {
```

Comment:
Should we check `event.isAnyPointerDown`? Either `pressOwner` isn't null in this case, or we missed previous presses, and then it is probably okay to call onOutsidePointerEvent (not sure)

Reply:
Yes. `pressOwner` will nulled after this function call if `event.isAnyPointerDown` is `false`.
So the check here is "it's a end of gesture that started without owner"